subcommand/lifecycle-sidecar/command.go

```diff
@@ -100,13 +103,12 @@ func (c *Command) Run(args []string) int {
 	//
 	// The loop will only exit when the Pod is shut down and we receive a SIGINT.
 	for {
-		for _, svc := range svcs {
-			err := c.consulClient.Agent().ServiceRegister(svc)
-			if err != nil {
-				logger.Error("failed to sync service", "id", svc.ID, "err", err)
-			} else {
-				logger.Info("successfully synced service", "id", svc.ID)
-			}
+		cmd := exec.Command(c.flagConsulLocation, "services", "register", c.flagServiceConfig)
```

Comment:
What about the ACL token? It'd be great if we could change the `lifecycle_sidecar.go` to use an env variable instead of the flag. That way, this command will just work™.

Reply:
Hmm, I didn't remove anything related to the ACL token. Looks like it might have been missing from before.

An environment variable would be cleaner for the lifecycle sidecar, but the connect injector deals exclusively in a token file, so it's an odd switch to make.

Reply:
Yeah, this was handled for us for free by the consul client because we instantiated it from the HTTP flags. We could set the `CONSUL_HTTP_TOKEN_FILE` env variable pointing to the file location instead.

Reply:
It would be easy to set the `CONSUL_HTTP_TOKEN_FILE` env variable in the lifecycle sidecar definition in the connect injector, but that means that the lifecycle sidecar process itself will not be able to independently handle cases where ACL tokens are enabled. Are we willing to make that trade-off? All of the other design decisions in the lifecycle sidecar process seem to indicate that it should be able to be run by itself.